exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/TraceTranslatorTest.java

```diff
@@ -0,0 +1,205 @@
+  }
+
+  @Test
+  public void toStatusProtoTest(){
+    io.opentelemetry.trace.Status myStatus = io.opentelemetry.trace.Status.OK.withDescription("Status description");
+    Status spanStatus = TraceTranslator.toStatusProto(myStatus);
+
+    // The int representation is 0 for canonical code "OK".
+    assertEquals(0,myStatus.getCanonicalCode().value());
+    assertEquals(io.opentelemetry.trace.Status.CanonicalCode.OK, myStatus.getCanonicalCode());
+    assertEquals("Status description", spanStatus.getMessage());
+  }
+  
+  @Test
+  public void toLinksProtoTest(){
+    List<io.opentelemetry.sdk.trace.data.SpanData.Link> links = new ArrayList<>();
+
+    TraceId traceIdOne = new TraceId(321, 123);
+    SpanId spanIdOne = new SpanId(12345);
+    TraceFlags traceFlagsOne = TraceFlags.builder().build();
+    TraceState traceStateOne = TraceState.builder().build();
+    SpanContext spanContextOne = SpanContext.create(traceIdOne, spanIdOne, traceFlagsOne, traceStateOne);
+    Attributes attributesOne = Attributes.newBuilder().setAttribute("key", "value").build();
+    io.opentelemetry.sdk.trace.data.SpanData.Link linkOne =  io.opentelemetry.sdk.trace.data.SpanData.Link.create(spanContextOne, attributesOne);
+
+    TraceId traceIdTwo = new TraceId(32473, 24893);
+    SpanId spanIdTwo = new SpanId(54321);
+    TraceFlags traceFlagsTwo = TraceFlags.builder().build();
+    TraceState traceStateTwo = TraceState.builder().build();
+    SpanContext spanContextTwo = SpanContext.create(traceIdTwo, spanIdTwo, traceFlagsTwo, traceStateTwo);
+    Attributes attributesTwo = Attributes.newBuilder().setAttribute("random string", "another random string").build();
+    io.opentelemetry.sdk.trace.data.SpanData.Link linkTwo =  io.opentelemetry.sdk.trace.data.SpanData.Link.create(spanContextTwo, attributesTwo);
+
+    links.add(linkOne);
+    links.add(linkTwo);
+    Span.Links finalLinks = TraceTranslator.toLinksProto(links, 2);
+
+    assertEquals(2, finalLinks.getLinkCount());
+    assertEquals(0, finalLinks.getDroppedLinksCount());
+
+    assertEquals(traceIdOne.toLowerBase16(), finalLinks.getLink(0).getTraceId());
+    assertEquals(spanIdOne.toLowerBase16(), finalLinks.getLink(0).getSpanId());
+    // The int representation is 0 for (Link.Type.TYPE_UNSPECIFIED), which is given in TraceTranslator.
+    assertEquals(0, finalLinks.getLink(0).getTypeValue());
+    assertTrue(finalLinks.getLink(0).getAttributes().containsAttributeMap("key"));
+
+    assertEquals(traceIdTwo.toLowerBase16(), finalLinks.getLink(1).getTraceId());
+    assertEquals(spanIdTwo.toLowerBase16(), finalLinks.getLink(1).getSpanId());
+    assertEquals(0, finalLinks.getLink(1).getTypeValue());
+    assertTrue(finalLinks.getLink(1).getAttributes().containsAttributeMap("random string"));
+    assertFalse(finalLinks.getLink(1).getAttributes().containsAttributeMap("key"));
+  }
+
+  @Test
+  public void getResourceLabelsTest(){
+    Map<String, String> nullResources = new HashMap<>();
```

Comment:
You can probably split this test into 2 separate tests - one for `nullResources` and one for non-empty set of resources. Currently you have 2 independent flows and it's hard to see how are they related to each other (they are not)

Reply:
Good point, done.